### PR TITLE
feat: sleeping mode avoided and orientation request bug on mobile is fixed

### PR DESCRIPTION
**Pull request by BusraTemizyurek**

### CODE REVIEW

File: src/index.ts [hunk 1]
```diff
@@ -1,5 +1,5 @@
 // TODO: Remove request-permission-polyfil before going to production
-import "./request-permission-polyfil";
+//import "./request-permission-polyfil";
```
Comment (altinokdarici) [inline src/index.ts:2]: why?

Comment (altinokdarici) [reply src/index.ts:2]: either remove it or uncomment it

File: src/wake-lock-manager.ts [hunk 2]
```diff
@@ -0,0 +1,20 @@
...
+    async requestWakeLock() {
+        try {
+            const curWakeLock = await navigator.wakeLock.request();
+            this._curWakeLock = curWakeLock;
+        } catch (err: any) {
+            console.log(`${err.name}, ${err.message}`);
+        }
+    }
+
+    async wakeLockOff() {
```
Comment (altinokdarici) [inline src/wake-lock-manager.ts:13]: make the fn name verb.

File: src/wake-lock-manager.ts [hunk 3]
```diff
@@ -0,0 +1,20 @@
...
+        if (this._curWakeLock) {
+            this._curWakeLock.release().then(() => {
+                this._curWakeLock = undefined;
+            });
```
Comment (altinokdarici) [inline src/wake-lock-manager.ts:17]: do not use `then`, use `await`.

File: src/request-permission-polyfil.ts [hunk 4]
```diff
@@ -1,6 +1,6 @@
-(DeviceMotionEvent as any).requestPermission = () => {
-    return new Promise((resolve) => {
-        (window as any).grantMotionPermission = () => resolve('granted');
-        (window as any).denyMotionPermission = () => resolve('denied');
-    });
-}
\ No newline at end of file
+// (DeviceMotionEvent as any).requestPermission = () => {
```
Comment (altinokdarici) [inline src/request-permission-polyfil.ts:1]: either remove the file completely or keep it uncommented. There is no point of having a commented file in the repo.

My suggestion is to uncomment this and keep it.

File: src/index.ts (hunk 1 above)
Comment (altinokdarici) [reply src/index.ts:2]: My suggestion is to remove this line instead of commenting.

File: src/main-page.ts [hunk 5]
```diff
@@ -60,6 +61,9 @@ export class MainPage implements Page {
     }
 
     private onClickModalReady(ev: ButtonMouseEvent) {
+        const wakeLock = new WakeLockManager();
+        wakeLock.requestWakeLock();
```
Comment (altinokdarici) [inline src/main-page.ts:65]: Why do you create a new wakeLockManager instance every time user clicks on this button? Shouldn't you create a single instance of it and re-use it across the app.

**Comment by altinokdarici** [comment]
I think you should create 3 PRs
1. Fix webpack config to allow all hosts 
2. Create a PR to fix orientation bug.
3. Create a PR to prevent sleeping.

They are 3 separate work items.

File: src/index.ts (hunk 1 above)
Comment (BusraTemizyurek) [reply src/index.ts:2]: I don't need it for now

File: src/wake-lock-manager.ts (hunk 2 above)
Comment (BusraTemizyurek) [reply src/wake-lock-manager.ts:13]: new fn name: releaseWakeLock()

File: src/wake-lock-manager.ts (hunk 3 above)
Comment (BusraTemizyurek) [reply src/wake-lock-manager.ts:17]: done

File: src/main-page.ts [hunk 6]
```diff
@@ -152,7 +182,9 @@ export class MainPage implements Page {
         rec.type = "button";
         rec.appendChild(circleIcon.node[0]);
         rec.classList.add("shadow-sm", "main-rec-button", "btn", "btn-outline-dark", "bg-secondary-subtle", "border", "border-4", "border-black", "rounded-circle");
-        rec.onclick = this.onRecordClick.bind(this);
+        rec.onclick = (ev) => {
+            this.onRecordClick(ev as ButtonMouseEvent);
+        }
```
[the remark this answers is not in the document]
Comment (altinokdarici) [reply src/main-page.ts:185]: ok, nevermind.

**Pull request closed without merging**